### ISSUE TITLE
Empty Moments are rendered incorrectly in transposed circuit diagrams

### ISSUE THREAD

**Issue by kevinsung** [quantumlib/Cirq#1315]
Empty Moments arise when optimizations leave existing Moments empty. For example:

Input:
```
import cirq

a, b = cirq.LineQubit.range(2)
circuit = cirq.Circuit.from_ops(
    cirq.Z.on_each([a, b]),
    cirq.Z.on_each([a, b]),
    cirq.CZ(a, b)
)
print('Original circuit:')
print(circuit.to_text_diagram(transpose=True))
print()

cirq.merge_single_qubit_gates_into_phased_x_z(circuit)
print('Optimized circuit:')
print(circuit.to_text_diagram(transpose=True))
print()

print('repr(circuit):')
print(repr(circuit))
```
Output:
```
Original circuit:
0 1
│ │
Z Z
│ │
Z Z
│ │
@─@
│ │

Optimized circuit:
0 1
│ │
  │
│ │
  │
│ │
@─@
│ │

repr(circuit):
cirq.Circuit(moments=[
    cirq.Moment(),
    cirq.Moment(),
    cirq.Moment(operations=[
        cirq.CZ.on(cirq.LineQubit(0), cirq.LineQubit(1)),
    ]),
])
```

As you can see, there are gaps in the wires in the diagram of the optimized circuit.

I think there's a case to be made that in this situation, the empty Moments shouldn't be rendered at all.

**Comment by vtomole** [comment]
How is this issue related to https://github.com/quantumlib/Cirq/issues/1131?

**Comment by kevinsung** [comment]
#1131 was resolved by PR #1193 , but that PR did not include any tests for transposed text diagrams.

**Comment by bryano** [comment]
See #1319.
